src/lib/request.ts

```diff
@@ -102,3 +92,47 @@ export async function request(
 
   return json.values;
 }
+
+function handleError(errorResponse: string | object) {
+  const error = new CiviCRMRequestError(errorResponse);
+
+  if (this.debug) {
+    console.error(error);
+    console.groupEnd();
+  }
+
+  throw error;
+}
+
+class CiviCRMRequestError extends Error {
+  name = "CiviCRMRequestError";
+  message: string;
+  detail: any;
+
+  parseError(error: string | { error_message: string }) {
+    if (typeof error === "object") {
+      return error;
+    }
+
+    try {
+      const object = JSON.parse(error);
+
+      if (object && typeof object === "object") {
+        return object;
```

Comment:
In both these return cases, there's an assumption that the objects contain an `error_message`. One way you could solve this is to move the ` || "CiviCRM request failed",` from line 126 to line 135.